domains/primitives/messenger/src/messages.rs

```diff
@@ -197,41 +197,88 @@ pub struct BlockInfo<Number, Hash> {
+pub struct ConsensusChainMmrLeafProof<BlockNumber, BlockHash, MmrHash> {
+    /// Consensus block info from which this proof was generated.
+    pub block_info: BlockInfo<BlockNumber, BlockHash>,
+    /// Encoded MMR leaf
+    pub opaque_mmr_leaf: EncodableOpaqueLeaf,
+    /// MMR proof for the leaf above.
+    pub proof: MmrProof<MmrHash>,
+}
+
+// TODO: remove this once the usage is unnecessary
+impl<BlockNumber, BlockHash, MmrHash> Default
+    for ConsensusChainMmrLeafProof<BlockNumber, BlockHash, MmrHash>
+where
+    BlockNumber: Default,
+    BlockHash: Default,
+{
+    fn default() -> Self {
+        Self {
+            block_info: BlockInfo::default(),
+            opaque_mmr_leaf: EncodableOpaqueLeaf(vec![]),
+            proof: MmrProof {
+                leaf_indices: vec![],
+                leaf_count: 0,
+                items: vec![],
+            },
+        }
+    }
+}
+
+impl<BlockNumber, BlockHash, MmrHash> Clone
+    for ConsensusChainMmrLeafProof<BlockNumber, BlockHash, MmrHash>
+where
+    BlockNumber: Clone,
+    BlockHash: Clone,
+    MmrHash: Clone,
+{
+    fn clone(&self) -> Self {
+        Self {
+            block_info: self.block_info.clone(),
+            opaque_mmr_leaf: EncodableOpaqueLeaf(self.opaque_mmr_leaf.0.clone()),
+            proof: self.proof.clone(),
+        }
+    }
+}
+
 /// Proof combines the storage proofs to validate messages.
 #[derive(Debug, Encode, Decode, Clone, Eq, PartialEq, TypeInfo)]
-pub struct Proof<BlockNumber, BlockHash, StateRoot> {
-    /// Consensus chain block info when proof was constructed
-    pub consensus_chain_block_info: BlockInfo<BlockNumber, BlockHash>,
-    /// State root of Consensus chain at above number and block hash.
-    /// This is the used to extract the message from proof.
-    pub consensus_chain_state_root: StateRoot,
-    /// Storage proof that src chain state_root is registered on Consensus chain.
-    /// This is optional when the src_chain is Consensus.
-    /// BlockNumber and BlockHash is used with storage proof to validate and fetch its state root.
-    pub domain_proof: Option<(BlockInfo<BlockNumber, BlockHash>, StorageProof)>,
+pub struct Proof<CBlockNumber, CBlockHash, MmrHash> {
+    /// Consensus chain MMR leaf proof.
+    pub consensus_chain_mmr_proof: ConsensusChainMmrLeafProof<CBlockNumber, CBlockHash, MmrHash>,
+    /// Storage proof that src domain chain's block is out of the challenge period on Consensus chain.
+    /// This is None when the src_chain is Consensus.
+    pub domain_proof: Option<StorageProof>,
     /// Storage proof that message is processed on src_chain.
     pub message_proof: StorageProof,
 }
 
-impl<BlockNumber: Default, BlockHash: Default, StateRoot: Default>
-    Proof<BlockNumber, BlockHash, StateRoot>
+impl<CBlockNumber: Default, CBlockHash: Default, MmrHash: Default>
+    Proof<CBlockNumber, CBlockHash, MmrHash>
 {
     #[cfg(feature = "runtime-benchmarks")]
     pub fn dummy() -> Self {
         Proof {
```

Comment:
Looks like Ning may have added this for benchamrking `relay_message` and `relay_message_response` as they are unsigned and proof verification is not done in those calls but rather done in `validate_unsigned`. So he may have opted to create a dummy here

Reply:
If that is the case, we'll want to explain in the comment why it is fine to do so. Still, I think for benchmarking purposes we will have to generate real proofs and this likely needs a TODO on it.

Though my preference would be to get rid of this and create a dummy value explicitly in the benchmark itself rather than sprinkling conditionally compiled dummy values like this throughout codebase.